### DISCUSSION TITLE
Build view buttons programatically

### DISCUSSION THREAD

**Discussion by jbaumgarten** [Rapptz/discord.py#9025] [Q&A] [answered]
Hi guys,

Is there any way to build dynamically a view with buttons?
How to link a button to his callback?

I found I can use something like:
```
class MultiPagedModalsView(discord.ui.View):
    def __init__(self, items):
        super().__init__()
        for item in items:
            button = discord.ui.Button(label=item['label'], style=discord.ButtonStyle.green)
            button.callback = self.callback
            self.add_item(button)

    async def callback(self, interaction: discord.Interaction):
       print("Test")
```

But how can I define which button is pressed?
Yours sincerely

**Comment by DenverCoder1** [comment, marked as answer]
There are many examples here you can follow:

https://github.com/Rapptz/discord.py/tree/master/examples/views

The [tic_tac_toe](https://github.com/Rapptz/discord.py/blob/master/examples/views/tic_tac_toe.py) example shows an organized way of creating a `Button` subclass with a callback and adding them dynamically to the view in the constructor.

Note that the button callback actually receives the button, so you can check its `label`, `custom_id`, or other attributes in the callback.

When using the decorator, `button` comes after the `interaction`. When constructing a button using `Button` class, the `self` parameter is the button.

**Reply by jbaumgarten** [reply]
I wanted to avoid create a specific class just for the callback
I found that something like

```
async def mycallback(interaction: discord.Interaction):
     print("Button pressed")

button = discord.ui.Button(...)
button.callback = mycallback
```
works

**Reply by jbaumgarten** [reply]
@Rapptz  why a down vote?
